src/local/room_xy.rs

```diff
@@ -229,6 +246,19 @@ impl RoomXY {
     }
 }
 
+impl PartialOrd for RoomXY {
```

Comment:
Strongly disagree with these impls existing, considering even in-game objects disagree on being row or column major (which this implicitly enforces).

Reply:
This was copied over from `Position`, purely because it allows for usage as a key in things like `BTreeMap` and that seems like a reasonable thing to keep parity with. If we want to tear it out here, we should tear it out there as well.

Reply:
Honestly, yes. Position is even worse, since on top of picking row/column major, you also have to decide if you're doing a lex order on the (room name, room xy) pair or treating the whole position as a single world xy.

Reply:
For position, there's really 6 different ways you could arrange the components into a 4-tuple and use the lex ordering to come up with a sensible Position ordering:
- (room_name.x, room_xy.x, room_name.y, room_xy.y)
- (room_name.y, room_xy.y, room_name.x, room_xy.x)
- (room_name.x, room_name.y, room_xy.x, room_xy.y)
- (room_name.x, room_name.y, room_xy.y, room_xy.x)
- (room_name.y, room_name.x, room_xy.x, room_xy.y)
- (room_name.y, room_name.x, room_xy.y, room_xy.x)

Reply:
I can tear this out if we really want to, and you can open up a PR for tearing out the `Position` implementation, but I'm not comfortable doing the latter in this PR. It feels out of scope for the intended goal of bringing `RoomXY` into approximate API parity with `Position`.